Build spx.so for Alpine Linux - php 8.5 issue

I have this Dockerfile which builds spx for Alpine Linux. It always worked fine up to php 8.4.15, but there's now an issue when trying to build for php 8.5.

```dockerfile
FROM alpine:3.22.2 AS build

# update repositories to edge
RUN printf "https://dl-cdn.alpinelinux.org/alpine/edge/main\nhttps://dl-cdn.alpinelinux.org/alpine/edge/community\n" > /etc/apk/repositories

# add testing repository
RUN printf "@testing https://dl-cdn.alpinelinux.org/alpine/edge/testing\n" >> /etc/apk/repositories

RUN apk update
RUN apk upgrade

# add c build tools
RUN apk add build-base

# add dev libraries
RUN apk add \
    php85-dev@testing \
    zlib-dev@testing

RUN apk add git

# fix ./configure "Cannot find php-config. Please use --with-php-config=PATH"
RUN ln -s /usr/bin/php-config85 /usr/bin/php-config

RUN git clone --depth 50 https://github.com/NoiseByNorthwest/php-spx.git

WORKDIR /php-spx

# checkout release - we need master for now
#RUN git checkout tags/v0.4.22

# build php-spx
RUN phpize85
RUN ./configure
RUN make
```

This is the error dump:

```txt
 => [build 10/14] RUN git clone --depth 50 https://github.com/NoiseByNorthwest/php-spx.git               2.3s
 => [build 11/14] WORKDIR /php-spx                                                                       0.2s
 => [build 12/14] RUN phpize85                                                                           1.6s
 => [build 13/14] RUN ./configure                                                                        2.2s
 => ERROR [build 14/14] RUN make                                                                         1.1s
------
 > [build 14/14] RUN make:
0.481 /bin/sh /php-spx/libtool --tag=CC --mode=compile cc -I. -I/php-spx -I/usr/include/php85 -I/usr/include/p
hp85/main -I/usr/include/php85/TSRM -I/usr/include/php85/Zend -I/usr/include/php85/ext -I/usr/include/php85/ex
t/date/lib  -DHAVE_CONFIG_H  -g -O2 -Werror -Wall -O3 -pthread -std=gnu90 -Wno-typedef-redefinition -g -D_GNU_
SOURCE    -DZEND_COMPILE_DL_EXT=1 -c /php-spx/src/php_spx.c -o src/php_spx.lo  -MMD -MF src/php_spx.dep -MT src/php_spx.lo
0.570 mkdir src/.libs
0.572  cc -I. -I/php-spx -I/usr/include/php85 -I/usr/include/php85/main -I/usr/include/php85/TSRM -I/usr/include/php85/Zend -I/usr/include/php85/ext -I/usr/include/php85/ext/date/lib -DHAVE_CONFIG_H -g -O2 -Werror -Wall
-O3 -pthread -std=gnu90 -Wno-typedef-redefinition -g -D_GNU_SOURCE -DZEND_COMPILE_DL_EXT=1 -c /php-spx/src/php
_spx.c -MMD -MF src/php_spx.dep -MT src/php_spx.lo  -fPIC -DPIC -o src/.libs/php_spx.o
0.686 In file included from /usr/include/php85/Zend/zend_compile.h:27,
0.686                  from /usr/include/php85/Zend/zend_modules.h:24,
0.686                  from /usr/include/php85/Zend/zend_API.h:25,
0.686                  from /usr/include/php85/main/php.h:35,
0.686                  from /php-spx/src/php_spx.h:23,
0.686                  from /php-spx/src/php_spx.c:29:
0.686 /usr/include/php85/Zend/zend_vm_opcodes.h:71:140: error: 'preserve_none' attribute directive ignored [-Werror=attributes]
0.686    71 | typedef const struct _zend_op *(ZEND_OPCODE_HANDLER_CCONV *zend_vm_opcode_handler_t)(struct _zend_execute_data *execute_data, const struct _zend_op *opline);
0.686       |
                                           ^~~~~~~~
1.030 cc1: note: unrecognized command-line option '-Wno-typedef-redefinition' may have been intended to silence earlier diagnostics
1.030 cc1: all warnings being treated as errors
1.033 make: *** [Makefile:216: src/php_spx.lo] Error 1
------
Dockerfile:42
--------------------
  40 |     RUN phpize85
  41 |     RUN ./configure
  42 | >>> RUN make
  43 |
  44 |     # start again with a new image
--------------------
ERROR: failed to build: failed to solve: process "/bin/sh -c make" did not complete successfully: exit code: 2

View build details: docker-desktop://dashboard/build/desktop-linux/desktop-linux/k2vgr1uhjig461p5sbg2ofcsm
```

From what I could figure out, the issue is that the compiler flags treat all warnings as errors, so removing the `-Werror` from the `CFLAGS` fixes the make and spx works fine as far as I could tell.

--- a/config.m4
+++ b/config.m4
@@ -19,7 +19,7 @@ if test "$PHP_SPX" = "yes"; then
     AC_DEFINE_UNQUOTED([SPX_HTTP_UI_ASSETS_DIR], [ "$PHP_SPX_ASSETS_DIR/web-ui" ], [path of web-ui assets directory])
     PHP_SUBST([PHP_SPX_ASSETS_DIR])
 
-    CFLAGS="$CFLAGS -Werror -Wall -O3 -pthread -std=gnu90"
+    CFLAGS="$CFLAGS -Werror -Wall -Wno-attributes -O3 -pthread -std=gnu90"
 
     # Disabling typedef-redefinition is required for:
     #   - macOS, see https://github.com/NoiseByNorthwest/php-spx/pull/270

